src/taskProvider.ts

```diff
@@ -23,6 +23,31 @@ type BuilderForLanguage = Partial<Record<TaskTargetLanguage, ExecBuilder>>;
 type TaskState = "enabled" | "disabled" | "unavailable";
 type BehaviorWhenAllTargetsSelected = "enable" | "disable" | "executeForEveryTarget";
 
+export interface ChecksList {
+  selected: string;
+  values: string[];
+}
+const validChecks: string[] = ["All", "manifest", "icons", "app_load_params", "makefile", "readme", "scan"];
+export const checks: ChecksList = {
+  selected:"All",
+  values: validChecks,
+};
+
+let checkSelectedEmitter: vscode.EventEmitter<string> = new vscode.EventEmitter();
+export const onCheckSelectedEvent: vscode.Event<string> = checkSelectedEmitter.event;
+
+export async function showChecks() {
```

Comment:
It feels weird to put the showChecks and everything related to the guideline enforcer directly into taskProvider...

Reply:
It is here because it is generic, and there is potential circular dependencies in other files. If we need more in the future, we'll put that in a dedicated file